libmscore/textline.cpp

```diff
@@ -187,7 +301,62 @@ QVariant TextLine::propertyDefault(Pid propertyId) const
             }
       }
 
+//---------------------------------------------------------
+//   setProperty
+//---------------------------------------------------------
 
+bool TextLine::setProperty(Pid id, const QVariant& v)
+      {
+      switch (id) {
+            case Pid::PLACEMENT:
+                  setPlacement(Placement(v.toInt()));
+                  break;
+            default:
+                  return TextLineBase::setProperty(id, v);
+            }
+      triggerLayout();
+      return true;
+      }
+
+//---------------------------------------------------------
+//   undoChangeProperty
+//---------------------------------------------------------
+
+void TextLine::undoChangeProperty(Pid id, const QVariant& v, PropertyFlags ps)
+      {
+      if (id == Pid::SYSTEM_FLAG) {
+            score()->undo(new ChangeTextLineProperty(this, v));
+            for (SpannerSegment* s : spannerSegments()) {
+                  score()->undo(new ChangeTextLineProperty(s, v));
+                  triggerLayout();
+                  }
+            MuseScoreCore::mscoreCore->updateInspector();
+            return;
+            }
+      initStyle();
```

Comment:
What is the purpose of this?  It causes any change to one property to reset all other properties to style defaults.